app/code/Magento/CheckoutAgreements/Block/Adminhtml/Agreement/Grid.php

```diff
@@ -17,15 +20,21 @@ class Grid extends \Magento\Backend\Block\Widget\Grid\Extended
      * @param \Magento\Backend\Helper\Data $backendHelper
      * @param \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\CollectionFactory $collectionFactory
      * @param array $data
+     * @param GridCollectionFactory $gridColFactory
      * @codeCoverageIgnore
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Backend\Helper\Data $backendHelper,
         \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\CollectionFactory $collectionFactory,
-        array $data = []
+        array $data = [],
+        GridCollectionFactory $gridColFactory = null
     ) {
-        $this->_collectionFactory = $collectionFactory;
+
+        $this->_collectionFactory = $gridColFactory
```

Comment:
It would be better define a new private property for grid collection, assign injected object and change all usages to new object (this step caused by backward compatibility, any extension may be used protected `$_collectionFactory` and expects concrete object type). 
The old property should be marked as deprecated.

Reply:
Done, I added private variable `gridCollectionFactory`